config/config.go

```diff
@@ -452,3 +485,13 @@ func (c *Configuration) CurrentCommitter() (name, email string) {
 	email, _ = c.Git.Get("user.email")
 	return
 }
+
+// RepositoryPermissions returns the permissions that should be used to write
+// files in the repository.
+func (c *Configuration) RepositoryPermissions() os.FileMode {
+	if c.mask == -1 {
+		c.setMask()
+	}
+
+	return os.FileMode(0666 & ^c.mask)
```

Comment:
What do you think about making this a function called `getMask()` or similar, like:

```go
func (c *Configuration) getMask() {
        c.maskOnce.Do(func() {
                c.mask = umask()
        })
        return c.mask
}
```

I think that that would allow us to forgo the `if c.mask == -1` conditional above, and rewrite this function as:

```go
return os.FileMode(0666 & ^c.getMask())
```

Reply:
Yeah, that would be tidier, I agree, and it would avoid us working with umask more than once.  I don't think we end up calling this in a multi-threaded context, but in case a refactor happens and we do, that would avoid any problems.